### PR TITLE
Backport bugfix PRs to the 6.1.x release branch

### PR DESCRIPTION
**Pull request by mdickinson**
This PR against the `release/6.1.x` branch:

- backports the changes from #1198, #1206, #1207, #1210, #1219, #1204, #1227, #1222, #1235 and #1250 to the 6.1.x release branch.
- updates the changelog for the above PRs (excluding PRs that are specific to the traits-stubs package)

The PR backports were all done using `git cherry-pick -x <commit_hash>`, with the commit hash referring to the merge commit for the corresponding PR against master.

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -1,6 +1,33 @@
...
+Documentation fixes
+~~~~~~~~~~~~~~~~~~~
+
+* Fix some spelling and grammar errors in the user manual. (#1210)
+* Fix description in README to match the one in the setup script. (#1219)
+* Update PyPI links and capitalization in README.rst. (#1250)
+* Fix user manual mentioning a nonexisting feature in metadata filter. (#1207)
+* Fix typo in comment in optional_dependencies. (#1235)
+
+
```
Comment (rahulporuri) [inline CHANGES.rst:30]: 1204 is missing from the changelog.

Comment (mdickinson) [reply CHANGES.rst:30]: From the PR description: "excluding PRs that are specific to the traits-stubs package"

**Comment by mdickinson** [comment]
Thanks for reviewing. The missing changelog entry is deliberate: that's a traits-stubs fix rather than a traits fix, so belongs in the traits-stubs changelog (which doesn't exist yet, but will need to be brought into existence when we next make a release of traits-stubs).

**Comment by rahulporuri** [comment]
Just to confirm, a 6.1.1 of traits also means a 6.1.1 of traits-stubs? If not, I'm not sure why the traits-stubs PR was backported.

**Comment by mdickinson** [comment]
> Just to confirm, a 6.1.1 of traits also means a 6.1.1 of traits-stubs?

No, not necessarily. We don't need to align traits releases and traits-stubs releases, but at some point a traits-stubs bugfix release will likely be made from the release/6.1.x branch.

> If not, I'm not sure why the traits-stubs PR was backported.

Because it'll eventually need to be backported (for the next time we make a traits-stubs release), and it's convenient to do all the existing backports at once.

**Comment by mdickinson** [comment]
> We don't need to align traits releases and traits-stubs releases

To be clear, we need to align for <major.minor> versions, but not for micro (bugfix) versions: the Traits API doesn't change in bugfix releases, so there's no necessity for an aligned traits-stubs release. Conversely, there may be traits-stubs fixes that we want to release in a traits-stubs bugfix release but that don't affect Traits: in that situation, we don't want to have to make a Traits release just to keep version numbers aligned.

**Comment by mdickinson** [comment]
And the real takeaway from this is that attempting to maintain multiple packages in a single repository is painful. It's much easier if each package has its own commit graph.

**Pull request merged**